### ISSUE TITLE
Cross-references inside document are broken with pandoc 2.8

### ISSUE THREAD

**Issue by N0rbert** [jgm/pandoc#5941]
I'm using pandoc under the hood of RMarkdown *bookdown* pipeline.

Below is the simplest *cross-reference.Rmd* document with single cross-reference:

```
---
title: Cross-reference
site: bookdown::bookdown_site
---

# Part 1 {#p1}

## Chapter 1 {#p1ch1}

Some text.

## Chapter 2 {#p1ch2}

Here we need to refer to chapter 1 as ``` `\@ref(p1ch1)` ```: see ch. \@ref(p1ch1).
```

I transform it to HTML using R language:

```
Rscript -e "bookdown::render_book('cross-reference.Rmd', 'bookdown::html_document2')"
```

I have double checked it - I have no problems with previous pandoc 2.7.3-1 version. The console output 

```
$ pandoc --version | head -n1
pandoc 2.7.3
$ Rscript -e "bookdown::render_book('cross-reference.Rmd', 'bookdown::html_document2')"
processing file: _main.Rmd
  |......................................................................| 100%
  ordinary text without R code

output file: _main.knit.md
/usr/bin/pandoc +RTS -K512m -RTS _main.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash+smart --output _main.html --email-obfuscation none --self-contained --wrap preserve --standalone --section-divs --template /home/user/R/x86_64-pc-linux-gnu-library/3.4/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --number-sections --variable 'theme:bootstrap' --include-in-header /tmp/RtmpVOCa6Z/rmarkdown-str4cdc44e194da.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --lua-filter /home/user/R/x86_64-pc-linux-gnu-library/3.4/rmarkdown/rmd/lua/pagebreak.lua --lua-filter /home/user/R/x86_64-pc-linux-gnu-library/3.4/rmarkdown/rmd/lua/latex-div.lua 

Output created: _main.html
[1] "/home/user/Documents/R/pandoc-2-8-bugs/_main.html"
```

and expected rendering is below:

>![273](https://user-images.githubusercontent.com/309456/69670486-153a1e00-10a5-11ea-8e83-46b6c0259f88.png)

But the newest pandoc 2.8 has a bug. The rendering is wrong (note two question marks below - "**??**") - see

>![28](https://user-images.githubusercontent.com/309456/69670571-41ee3580-10a5-11ea-9da4-a2bcf2062e52.png)

and the console output is below:

```
$ pandoc --version | head -n1
pandoc 2.8
$ Rscript -e "bookdown::render_book('cross-reference.Rmd', 'bookdown::html_document2')"
processing file: _main.Rmd
  |......................................................................| 100%
  ordinary text without R code

output file: _main.knit.md
/usr/bin/pandoc +RTS -K512m -RTS _main.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash+smart --output _main.html --email-obfuscation none --self-contained --wrap preserve --standalone --section-divs --template /home/user/R/x86_64-pc-linux-gnu-library/3.4/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --number-sections --variable 'theme:bootstrap' --include-in-header /tmp/RtmpDtVKpd/rmarkdown-str49e574e87c73.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --lua-filter /home/user/R/x86_64-pc-linux-gnu-library/3.4/rmarkdown/rmd/lua/pagebreak.lua --lua-filter /home/user/R/x86_64-pc-linux-gnu-library/3.4/rmarkdown/rmd/lua/latex-div.lua 

Output created: _main.html
[1] "/home/user/Documents/R/pandoc-2-8-bugs/_main.html"
Warning message:
The label(s) p1ch1 not found 
```
Note the warning `The label(s) p1ch1 not found ` in the output above.

The problem is caused only by pandoc 2.8. As all other related components remain the same.

All often-used writers such as HTML, EPUB, ODT and DOCX are affected. But the PDF is not affected.

Please test this approach on your side and fix the bug if you can confirm it.

**Comment by jgm** [comment]
Please report this issue to bookdown developers.  The syntax you're using to generate a chapter reference is not part of pandoc itself; an external tool must be resolving this.
